lib/screens/movie_home_page.dart

```diff
@@ -1,15 +1,24 @@
 import 'package:flutter/material.dart';
-import '/custom_widgets/movie_body.dart';
-import '/custom_widgets/movie_app_bar.dart';
+import '/movie_widgets/movie_app_bar.dart';
+import '/movie_widgets/movie_body.dart';
+
 import '/data_classes/movie.dart';
 
 class MovieHomePage extends StatelessWidget {
-  MovieHomePage({super.key});
+  const MovieHomePage({super.key});
 
-  final Movie movie = Movie.defaultMovie();
+  static const String routeName = "/movie";
+  static const String nullError = "We had an issue loading the movie";
 
   @override
   Widget build(BuildContext context) {
+    final arg = ModalRoute.of(context)?.settings.arguments;
+    if (arg is! Movie) {
+      return const Text(
+        nullError,
+      );
+    }
```

Comment:
To void this validation here, you can use the body, to show or the error text or the corresponding movie, like this:
`Scaffold(
     appBar: ....,
     body: movie != null ? MovieBody(movie: movie) : Text(errorMsg)
     `

Reply:
reviewed